bpo-30414: multiprocessing.Queue._feed do not break from main loop on exc

grzgrzgrz3 opened the pull request:
Queue background running thread was not handling exceptions correctly.
Any exception occurred inside thread (putting unpickable object) cause
feeder to finish running. After that every message put into queue is
silently ignored.

grzgrzgrz3 committed:
bpo-30414: multiprocesing.Queue._feed do not break from main loop on exc

Queue background running thread was not handling exceptions correctly.
Any exception occurred inside thread (putting unpickable object) cause
feeder to finish running. After that every message put into queue is
silently ignored.

pitrou commented on Lib/test/_test_multiprocessing.py:
```diff
@@ -752,6 +752,16 @@ def test_timeout(self):
         # Windows (usually 15.6 ms)
         self.assertGreaterEqual(delta, 0.170)
 
+    def test_queue_feder_donot_stop_onexc(self):
```
There's a typo here ("feder"). Also can you add a comment pointing out to the issue number?

pitrou commented on Lib/test/_test_multiprocessing.py:
```diff
@@ -752,6 +752,16 @@ def test_timeout(self):
+            def __reduce__(self):
+                raise AttributeError
+        with test.support.captured_stderr():
+            q = multiprocessing.Queue()
```
Why not `self.Queue()` as in most other tests?

grzgrzgrz3 replied:
This test is only for multiprocessing.Queue, manager and threading queues implementations do not start feeder. 
I will use self.Queue and skip test for manager and threading.

grzgrzgrz3 committed:
bpo-30414: multiprocesing.Queue._feed do not break from main loop on exc

Queue background running thread was not handling exceptions correctly.
Any exception occurred inside thread (putting unpickable object) cause
feeder to finish running. After that every message put into queue is
silently ignored.

pitrou committed:
Merge branch 'master' of https://github.com/python/cpython into pr_1683
